.gitignore

```diff
@@ -1,9 +1,8 @@
-# example binaries
-examples/dial/example_dial
-examples/echo/example_echo
+# Please try to keep this file as empty as possible,
```

Comment:
Can you explain the logic here a little bit? glancing over other `.gitignore` files from viam repos, they seem to be comfortable with a broader list. What is the argument against doing so?

Happy to discuss offline if easier!

Reply:
The goal is for all build state to live entirely under the `build` directory, so it shouldn't be necessary to ignore many things once you have ignored the whole build directory. There also shouldn't be much need to ignore things like IDE configuration files (e.g. `.dir_locals.el` for emacs) in the project specific gitignore, since if you as a developer are using a particular IDE or tool you can just put the relevant matches into your personal global gitignore file and then it works across all your projects without needing to customize each one. And the file often becomes append only: once something gets added to the project gitignore file it never gets removed because it becomes unclear why it was added or for whom. So it ends up accumulating things forever. That's an opinionated take, obviously.

Reply:
@stuqdog - Resolving this thread about `.gitignore`, please re-open if you want to continue the discussion.